code/modules/mob/living/simple_animal/hostile/megafauna/_megafauna.dm

```diff
@@ -118,24 +118,48 @@
 			if(!client && ranged && ranged_cooldown <= world.time)
 				OpenFire()
 
-			if(L.health <= HEALTH_THRESHOLD_DEAD && HAS_TRAIT(L, TRAIT_NODEATH)) //Nope, it still gibs yall
+			if(L.health <= HEALTH_THRESHOLD_DEAD && HAS_TRAIT(L, TRAIT_NODEATH)) //Nope, it still kills yall
 				devour(L)
 		else
 			devour(L)
 
 /// Devours a target and restores health to the megafauna
 /mob/living/simple_animal/hostile/megafauna/proc/devour(mob/living/L)
-	if(!L)
+	if(!L || HAS_TRAIT(L, TRAIT_GUTTED))
 		return FALSE
-	visible_message(
-		span_danger("[src] devours [L]!"),
-		span_userdanger("You feast on [L], restoring your health!"))
+	celebrate_kill()
 	if(!is_station_level(z) || client) //NPC monsters won't heal while on station
 		adjustBruteLoss(-L.maxHealth/2)
 	L.investigate_log("has been devoured by [src].", INVESTIGATE_DEATHS)
-	L.gib()
+	var/mob/living/carbon/carbonTarget = L
+	if(istype(carbonTarget))
+		qdel(L.get_organ_slot(ORGAN_SLOT_LUNGS))
+		qdel(L.get_organ_slot(ORGAN_SLOT_HEART))
+		qdel(L.get_organ_slot(ORGAN_SLOT_LIVER))
+	L.adjustBruteLoss(500)
+	L.death() //make sure they die
+	L.RegisterSignal(L, COMSIG_MOB_STATCHANGE, PROC_REF(on_gutted_statchange))
+	ADD_TRAIT(L, TRAIT_GUTTED, TRAIT_GENERIC)
+	LoseTarget()
 	return TRUE
 
+/mob/living/simple_animal/hostile/megafauna/proc/celebrate_kill(mob/living/L)
+	visible_message(
+		span_danger("[src] devours [L]'s organs!"),
+		span_userdanger("You feast on [L]'s organs, restoring your health!"))
+
+/// When the mob's stat changes after being gutted by a megafauna (they're revived, etc)
+/mob/living/proc/on_gutted_statchange(mob/living/L, new_stat)
+	SIGNAL_HANDLER
+	REMOVE_TRAIT(L, TRAIT_GUTTED, TRAIT_GENERIC)
+	UnregisterSignal(L, COMSIG_MOB_STATCHANGE)
```

Comment:
IMO this is better handled by a status effect than this, reduces chance of signal collision

Reply:
how would I turn this into a status effect? I'm not familiar with it.

Reply:
```dm
/datum/status_effect/gutted
    id = "gutted"
    alert_type = null
    duration = -1
    tick_interval = -1

/datum/status_effect/gutted/on_apply()
    RegisterSignal(owner, COMSIG_MOB_STATCHANGE, PROC_REF(stop_gutting))
    return TRUE

/datum/status_effect/gutted/on_remove()
    UnregisterSignal(owner, COMSIG_MOB_STATCHANGE)

/datum/status_effect/gutted/proc/stop_gutting()
    SIGNAL_HANDLER
    qdel(src)
```
```dm
    dead_guy.apply_status_effect(/datum/status_effect/gutted)
```
```dm
    if(dead_guy.has_status_effect(/datum/status_effect/gutted)
        return FALSE
```
This has the added benefit of being able to add a visual effect or examine message to a gutted mob.

I suggest adding a red tint to gutted mobs and adding an examine message like "They appear to have been ravaged.", this just makes it clearer just what happened to the guy.

Reply:
>I suggest adding a red tint to gutted mobs and adding an examine message like "They appear to have been ravaged.", this just makes it clearer just what happened to the guy.

I'm pretty sure the fact they're a corpse next to a megafauna with tons of damage and no heart tells enough